module/documents/active-effect.mjs

```diff
@@ -414,25 +414,28 @@ export default class ActiveEffect5e extends ActiveEffect {
    * @param {object} options  Options passed to the effect creation.
    */
   async createRiderEnchantments(options) {
-    const origin = await fromUuid(this.origin)?? game.messages.get(options?.chatMessageOrigin)?.getAssociatedItem();
```

Comment:
This is something we lost when we moved the enchantment data out of flags. We cannot get the rider information from the AE anymore, we have to look it up on the original enchantment Activity, but to do that we need to know the `enchantmentProfile` that was used. We can't get this information at all without `options.chatMessageOrigin`.

So we lose the ability to apply enchantments by drag-dropping the enchantment AE onto an Item. They must be applied via the chat card now. If they don't have riders then I suppose it's still possible to apply via drag-drop.

I don't know of an easy way around this limitation where the `origin` points to the Item rather than the Activity (in the case of drag-dropping the AE), since the relationship from enchantment Activities to AEs is many-to-many now.

Reply:
Could we override drag-drop and add the id of an activity from the item?

Reply:
There could be multiple activities that reference the same AE, so we can't know which one to use in the general case.

Reply:
Using the first found might be "good enough" in the meantime?

Reply:
Yeah, you're right, it's better than nothing